tests/README.md

```diff
@@ -0,0 +1,47 @@
+# s2n-tls testing
+s2n-tls has a wide variety of tests covering a wide variety of different functionality. This document serves as a "quick reference", providing an overview of all the testing that s2n-tls does and linking to additional documentation.
+
+## Unit Tests
+_Is the internal functionality of s2n-tls correct?_
+
+These test the correctness of low-level s2n-tls functions. These functions are commonly private. We have unit tests for both the [C library](unit/) and the [rust bindings](../bindings/rust). 
+
+### Sanitizers
+The C unit tests are additionally run with the following sanitizers enabled
+- Valgrind: used to detect memory errors
+- Address Sanitizer (ASAN): used to detect memory errors
+- Thread Sanitizer (TSAN): used to detect race conditions
+
+## Integration Tests
+_Is the functionality of s2n-tls correct and RFC-compliant?_
+
+Our integration tests involve interoperation with other TLS implementations, because testing against another implementation improves confidence that s2n-tls has implemented the feature correctly and is compliant with relevant RFCs.
+
+s2n-tls integration tests are run using our [integrationv2](integrationv2/README.md) framework. Tests are implemented using "client" and "server" executables from various implementations. The client and server interactions are coordinated with [pytest](https://docs.pytest.org/en/stable/).
+
+## Fuzz Tests
+Fuzz Tests provide additional confidence in s2n-tls correctness by testing s2n-tls functionality under a wide vareity of inputs.
+
+s2n-tls [fuzz tests](fuzz/Readme.md) use LibFuzzer as the fuzzing engine. Fuzzing is generally only applied to functions that take external input.
+
+## Formal Methods
+s2n-tls includes a variety of formal methods which are used to _prove_ that s2n-tls has certain behaviors.
+
+### CBMC
+> CBMC verifies memory safety (which includes array bounds checks and checks for the safe use of pointers), checks for various further variants of undefined behavior, and user-specified as­ser­tions.
```

Comment:
Might be worth linking to the project for more information, especially if it's quoted: https://www.cprover.org/cbmc/